### PR TITLE
COST-2107: Remove unnecessary AWS Partables.

### PR DESCRIPTION
**Pull request by myersCody**
## HOW TO TEST
1. Check out main and load the database with aws data
2. Log into the db and check to make sure the partable tables are there:
```
postgres=# select table_schema, 
       table_name 
  from information_schema.tables 
 where table_name in (
           'reporting_aws_compute_summary_by_region_p', 
           'reporting_aws_compute_summary_by_service_p', 
           'reporting_aws_storage_summary_by_region_p', 
           'reporting_aws_storage_summary_by_service_p'
       );

 table_schema |                 table_name
--------------+--------------------------------------------
 template0    | reporting_aws_compute_summary_by_region_p
 template0    | reporting_aws_compute_summary_by_service_p
 template0    | reporting_aws_storage_summary_by_region_p
 template0    | reporting_aws_storage_summary_by_service_p
 acct10001    | reporting_aws_compute_summary_by_service_p
 acct10001    | reporting_aws_compute_summary_by_region_p
 acct10001    | reporting_aws_storage_summary_by_region_p
 acct10001    | reporting_aws_storage_summary_by_service_p
(8 rows)
```
3. Check out this branch, and run the migrations
```
make run-migrations
```
4) Check the db again
```
postgres=# select table_schema,
       table_name
  from information_schema.tables
 where table_name in (
           'reporting_aws_compute_summary_by_region_p',
           'reporting_aws_compute_summary_by_service_p',
           'reporting_aws_storage_summary_by_region_p',
           'reporting_aws_storage_summary_by_service_p'
       );
 table_schema | table_name
--------------+------------
(0 rows)
```

### CODE REVIEW

File: koku/reporting/migrations/0213_aws_rm_unused_partables.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    operations = [
+        migrations.RemoveField(model_name="awscomputesummarybyservicep", name="account_alias"),
+        migrations.RemoveField(model_name="awscomputesummarybyservicep", name="organizational_unit"),
+        migrations.RemoveField(model_name="awscomputesummarybyservicep", name="source_uuid"),
+        migrations.RemoveField(model_name="awsstoragesummarybyregionp", name="account_alias"),
+        migrations.RemoveField(model_name="awsstoragesummarybyregionp", name="organizational_unit"),
+        migrations.RemoveField(model_name="awsstoragesummarybyregionp", name="source_uuid"),
+        migrations.RemoveField(model_name="awsstoragesummarybyservicep", name="account_alias"),
+        migrations.RemoveField(model_name="awsstoragesummarybyservicep", name="organizational_unit"),
+        migrations.RemoveField(model_name="awsstoragesummarybyservicep", name="source_uuid"),
```
Comment (Red-HAP) [inline koku/reporting/migrations/0213_aws_rm_unused_partables.py:18]: Why RemoveField when you're removing the model? Am I missing something?

File: koku/reporting/migrations/0213_aws_rm_unused_partables.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# Generated by Django 3.1.13 on 2022-01-05 20:47
+from django.db import migrations
+
```
Comment (Red-HAP) [inline koku/reporting/migrations/0213_aws_rm_unused_partables.py:3]:
```suggestion
from koku.database import get_model

PartitionedTables = get_model("partitioned_tables")
TARGET_PARTABLES = (
    "reporting_aws_compute_summary_by_region_p",
    "reporting_aws_compute_summary_by_service_p",
    "reporting_aws_storage_summary_by_region_p",
    "reporting_aws_storage_summary_by_service_p",
)

def drop_partitions(app, schema_editor):
    PartitionedTables.objects.filter(partition_of_table_name__in=TARGET_TABLES).delete()

```

File: koku/reporting/migrations/0213_aws_rm_unused_partables.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+class Migration(migrations.Migration):
+
+    dependencies = [("reporting", "0212_auto_20211203_1640")]
+
+    operations = [
```
Comment (Red-HAP) [inline koku/reporting/migrations/0213_aws_rm_unused_partables.py:9]:
```suggestion
    operations = [
        migrations.RunPython(code=drop_partitions, reverse_code=migrations.RunPython.noop),
```

**Review by Red-HAP — CHANGES_REQUESTED** [review]

File: koku/reporting/migrations/0213_aws_rm_unused_partables.py (hunk 1 above)
Comment (myersCody) [reply koku/reporting/migrations/0213_aws_rm_unused_partables.py:18]: To be honest I autogenerated this migration. I thought it was weird to drop those fields first too, but then I noticed that all of these fields were foreign keys. I think this is Django's way to handle potential cascading deletes on the foreign keys values before the model is deleted. That or to check if there are any foreign key constraints that would prevent the models from deleting.

**Comment by myersCody** [comment]
Successful Smokes: https://ci.ext.devshift.net/job/project-koku-koku-pr-check/3471/

**Comment by myersCody** [comment]
Moved to:  https://github.com/project-koku/koku/pull/3408

**Pull request closed without merging**